File: arr_rotate.cpp
#include<iostream>
#include<vector>

using namespace std;

void rotate(vector<vector<int>>& matrix) {
	int temp;
	for(int i=0;i<matrix.size();i++){
		
	}	
}

void print(vector<vector<int>>& matrix) {
	for(int i=0;i<5;i++){
		for(int j=0;j<5;j++){
			cout<<matrix[i][j]<<endl;
		}
	}
}

int main(int argc,char** argv){
	vector<vector<int>> matrix;

	for(int i=0;i<5;i++){
		for(int j=0;j<5;j++){
			matrix[i].push_back(j);
		}
	}

	print(matrix);
}

File: arr_rotate.java
public void rotate(int[][] matrix) {
	int temp;
	int dest_j = 0;

	for(int i=0;i<matrix.length;i++){
		for(int j=0;j<matrix[i].length;j++){
			dest_j = j + matrix[i].length - 1;
			temp = matrix[i][dest_j];
			matrix[i][dest_j] = matrix[i][j];
		}
	}        
}
